MathLib/Curve/PiecewiseLinearCurve.h

```diff
@@ -0,0 +1,44 @@
+/**
+ * \copyright
+ * Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ * \file   PiecewiseLinearCurve.h
+ *
+ * Created on November 11, 2016, 10:49 AM
+ */
+
+#ifndef OGS_PIECE_WISE_LINEAR_CURVE_H
+#define OGS_PIECE_WISE_LINEAR_CURVE_H
+
+#include "MathLib/InterpolationAlgorithms/PiecewiseLinearInterpolation.h"
+
+namespace MathLib
+{
+class PiecewiseLinearInterpolation;
+
+class PiecewiseLinearCurve final : public PiecewiseLinearInterpolation
+{
+public:
+    /**
+     * @ x x coordinates of curve points
+     * @ y y coordinates of curve points
+     */
+    PiecewiseLinearCurve(std::vector<double>&& x, std::vector<double>&& y)
+        : PiecewiseLinearInterpolation(std::move(x), std::move(y), false)
+    {
```

Comment:
see my last comment.  How do you think about the idea to create two classes
PiecewiseLinearCurve
PiecewiseLinearMonotonicCurve

With that
getDerivative will be moved to PiecewiseLinearCurve, and PiecewiseLinearMonotonicCurve is derived from PiecewiseLinearCurve.